Write ecology and intro (#4)
SW814F20/Report: report/PreProject/Ecology.tex

\section{Ecology Interfaces}
This section will cover the process in finding the Ecology Interfaces, which we might use in the project, and rank them.

The idea in Ecology is to get an idea of external services, external artifacts, external repositories and external people, which might help us to solve the problem at hand.
After we have gathered all of the points, we should filter them, meaning that we need to use some sort of metrics to rank them, to get an overview of which points might be better suited for us, than others.
This should not be interpreted as, we do not use them or do not care about them, but more as we know they are there, and they might be useful, however, not at the time.

> Mentioned before we have four categories we need to find out, which are all externals.

Reviewer: ```suggestion
Mentioned before we have four categories we need to examine, which are all externals.
```

> The \textbf{external service} is services that we can use, without modification, which in the end, will make our solution easier to do, as we need to do less work.\\

Reviewer: ```suggestion
The \textbf{external service} is services that we can use, without modification, which in the end, will make our solution easier to realize, as we need to do less work.\\
```

In our process, we look at one category at a time and then brainstormed to get ideas of what should be there.
When we got an idea, we would talk about it, and why it should be there.

For the external services, we have these points:

\begin{itemize}
    \item Mobile Platform
    \item Email Platform
    \item Web Platform
    \item Board
    \item Calendar
    \item Chat 
\end{itemize}

For the external artifacts, we have these points

\begin{itemize}
    \item Mobile
    \item Database
    \item Smartwatch
\end{itemize}

For the external repositories, we have these points

\begin{itemize}
    \item GitHub
\end{itemize}

For the external people, we have these points

\begin{itemize}
    \item Mediator
    \item Customer
    \item Developer
    \item Agile coach
\end{itemize}

After we got all of these points, we need to filter them.
This can be done in different ways.
We can use \textbf{Single Idea Evaluation} and \textbf{Comparative Idea Evaluation}.
As the name states, the single idea is to evaluate a single idea, and comparative is where we take ideas that are alternatives to each other and see what idea seems to be the best.
We will use both of these evaluation techniques.

> We use the single idea evaluation to get a better understanding of the idea, and then we will use comparative idea evaluation, to rank the different ideas against each other, to get this ranking.

Reviewer: ```suggestion
We use the single idea evaluation to get a better understanding of the idea, and then we will use comparative idea evaluation, to rank the different ideas against each other.
```
